src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -633,8 +639,7 @@ static int parse_request(uint8_t code, uint8_t len, const void *option, void *us
                 break;
         case SD_DHCP_OPTION_MAXIMUM_MESSAGE_SIZE:
                 if (len == 2)
-                        req->max_optlen = be16toh(*(be16_t*)option) -
-                                          - sizeof(DHCPPacket);
+                        req->max_optlen = unaligned_read_be16(option) + sizeof(DHCPPacket);
```

Comment:
Whilst you are at it, it probably makes sense to verify that the received value is at least DHCP_DEFAULT_MIN_SIZE, and drop the option (and log) otherwise. The client would only hurt themselves by sending a bogus value here, but we should try to continue if at all possible of course.